modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/shp/PolygonHandler.java

```diff
@@ -526,7 +525,7 @@ public void write(ByteBuffer buffer, Object geometry) {
             // m values
             values.forEach(
                     x -> {
-                        buffer.putDouble(!Double.isNaN(x) ? x : -10E40);
+                        buffer.putDouble(Double.isNaN(x) ? -10E40 : x);
```

Comment:
Is this a specific magic number or will any large negative number do e.g. Negative Infinity? Could we use a constant?

Reply:
I've no idea, I just reverted the negative cause it hurts my eyes

Reply:
To answer your question: Any floating point number smaller than `–10^38` is considered by a shapefile reader to represent a **"no data"** value

From the spec:
```
Floating point numbers must be numeric values. Positive infinity, negative infinity, and
Not-a-Number (NaN) values are not allowed in shapefiles. Nevertheless, shapefiles
support the concept of "no data" values, but they are currently used only for measures.
38
Any floating point number smaller than –10^38 is considered by a shapefile reader to
represent a "no data" value.
```